### PR TITLE
Tab fix in yaml-editor edit card

### PR DESCRIPTION
**Pull request by yosilevy**
Implemented behavior to override the function that calculates focusable elements which fails since it doesn't use the shadowRoot.
Added new ha-paper-dialog which uses the behavior.
Used only in card editor. If approved should be used in all dialogs since now tabs don't work at all.
Also updated yaml mode in codemirror to use the proper indentMore/less and not the silly behavior it had now (auto mode which fails miserably).

### CODE REVIEW

File: src/panels/lovelace/components/dialog/ha-iron-focusables-helper.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/*
+  Fixes issue with not using shadow dom properly in iron-overlay-behavior/icon-focusables-helper.js
+*/
+
+// import "@polymer/polymer/polymer-legacy.js";
```
Comment (balloob) [inline src/panels/lovelace/components/dialog/ha-iron-focusables-helper.js]: Stale  comment

File: src/panels/lovelace/components/dialog/ha-iron-focusables-helper.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+/*
+  Fixes issue with not using shadow dom properly in iron-overlay-behavior/icon-focusables-helper.js
+*/
```
Comment (balloob) [inline src/panels/lovelace/components/dialog/ha-iron-focusables-helper.js]: Can you include the original source code license

**Review by balloob — APPROVED** [review]
We should probably use this everywhere where we use paper-dialog?

**Comment by yosilevy** [comment]
Yes I wrote that above:

> Used only in card editor. If approved should be used in all dialogs since now tabs don't work at all.

**Comment by yosilevy** [comment]
Travis fails with:
error  '@polymer/iron-overlay-behavior' should be listed in the project's dependencies. Run 'npm i -S @polymer/iron-overlay-behavior' to add it  import/no-extraneous-dependencies

@balloob Are you doing that?

**Comment by yosilevy** [comment]
Fixed. Please approve and we will fix all other dialogs in a later commit. It requires testing time that I don't have right now. Need to modify all associated styles as well (there are references to paper-dialog)

File: src/panels/lovelace/editor/card-editor/hui-edit-card.ts [hunk 3]
```diff
@@ -79,7 +80,7 @@ export class HuiEditCard extends LitElement {
   private _cardType?: string;
 
   private get _dialog(): PaperDialogElement {
```
Comment (balloob) [inline src/panels/lovelace/editor/card-editor/hui-edit-card.ts]: Wrong return value

**Review by balloob — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -34,6 +34,7 @@
     "@polymer/iron-input": "^3.0.1",
     "@polymer/iron-label": "^3.0.1",
     "@polymer/iron-media-query": "^3.0.1",
+    "@polymer/iron-overlay-behavior": "^3.0.0-pre.27",
```
Comment (balloob) [inline package.json]: 3.0.2 is the latest version https://www.npmjs.com/package/@polymer/iron-overlay-behavior

Comment (yosilevy) [reply package.json]: Yeah well let me see 1 Travis build passing and I'll change this to later versions and see if it works. So far this hasn't happened. Yarn.lock has this pre.27 version and when I do yarn add it adds the new version but doesn't remove the old one...

**Pull request merged**